clients/src/test/java/org/apache/kafka/common/requests/RequestResponseTest.java

```diff
@@ -292,10 +296,18 @@ private AbstractRequestResponse createMetadataResponse() {
         return new MetadataResponse(Arrays.asList(node), allTopicMetadata);
     }
 
-    private AbstractRequest createOffsetCommitRequest() {
+    private AbstractRequest createOffsetCommitRequest(int version) {
         Map<TopicPartition, OffsetCommitRequest.PartitionData> commitData = new HashMap<>();
         commitData.put(new TopicPartition("test", 0), new OffsetCommitRequest.PartitionData(100, ""));
-        return new OffsetCommitRequest("group1", 100, "consumer1", 1000000, commitData);
+        commitData.put(new TopicPartition("test", 1), new OffsetCommitRequest.PartitionData(200, null));
+        if (version == 0) {
+            return new OffsetCommitRequest("group1", commitData);
+        } else if (version == 1) {
+            return new OffsetCommitRequest("group1", 100, "consumer1", commitData);
+        } else if (version == 2){
```

Comment:
Yeah, sorry about that. I'll fix in the morning. I also need to add a test case which actually checks the null metadata case.